### DISCUSSION TITLE
Filename / Path Length for manifests

### DISCUSSION THREAD

**Discussion by timgmfc** [fluxcd/flux2#2229] [Q&A]
Is there a documented filename/path length limitation on the source or kustomize controller?  I've had a few manifests that didn't sync until I moved them up the tree a few levels.  There are no explicit errors or log messages about skipping these files in either controller, they are just not created.

Examples:
Does not sync: /namespace/folder/[folder recursion due to `helm template` output]/manifest.yaml (does not create) (in this case,138 characters long)

Successful sync: /namespace/folder/fewer folder recursions/manifest.yaml (creates) (in this case, 103 characters long)

**Comment by kingdonb** [comment]
We brought this post up this week at our Bug Scrub, and at least one of us had seen this issue occurring live but didn't follow up at the time.

That is a surprising issue, but it sounds easy to reproduce. If you're still seeing this on the last Flux 0.26.x release, we will probably want to convert this discussion into an issue so it is more likely to get follow-up.

If there is a file path/name limit, then I think that's a bug. I found this section of source code which seems to indicate we should support file paths longer than 260 characters, overcoming some limitation at that length on certain platforms:

https://github.com/fluxcd/source-controller/blob/273f8b155eb267e981120c38cdcccc0440659a2e/internal/fs/fs.go#L217-L222

So, could you please re-file this as a bug? 😄 Including information about your kubernetes cluster and what architecture/platform it is running on will help, as I didn't find this issue reported before, it may also be some specific issue related to your platform (like the one mentioned by the code that works around, above.) or host filesystem or any other detail you can provide which might be relevant
